diff --git a/GasPlanet.php b/GasPlanet.php
index df9c6b0..410a7a7 100644
--- a/GasPlanet.php
+++ b/GasPlanet.php
@@ -2,6 +2,11 @@
 
 class GasPlanet implements PlanetInterface
 {
+    const MATERIAL_AMMONIA = 'ammonia';
+    const MATERIAL_HYDROGEN = 'hydrogen';
+    const MATERIAL_HELIUM = 'helium';
+    const MATERIAL_METHANE = 'methane';
+
     private $diameter;
 
     private $mainElement;
@@ -21,12 +26,12 @@ class GasPlanet implements PlanetInterface
     {
         // a "fake" map of elements to colors
         switch ($this->mainElement) {
-            case 'ammonia':
+            case GasPlanet::MATERIAL_AMMONIA:
                 return '663300';
-            case 'hydrogen':
-            case 'helium':
+            case GasPlanet::MATERIAL_HYDROGEN:
+            case GasPlanet::MATERIAL_HELIUM:
                 return 'FFFF66';
-            case 'methane':
+            case GasPlanet::MATERIAL_METHANE:
                 return '0066FF';
             default:
                 return '464646';
diff --git a/index.php b/index.php
index c5000ed..01e239f 100644
--- a/index.php
+++ b/index.php
@@ -4,8 +4,8 @@ require __DIR__ . '/PlanetRenderer.php';
 require __DIR__ . '/PlanetInterface.php';
 require __DIR__ . '/GasPlanet.php';
 
-$planet1 = new GasPlanet('ammonia', 85);
-$planet2 = new GasPlanet('methane', 72);
+$planet1 = new GasPlanet(GasPlanet::MATERIAL_AMMONIA, 85);
+$planet2 = new GasPlanet(GasPlanet::MATERIAL_METHANE, 72);
 
 $renderer = new PlanetRenderer();
 
