diff --git a/index.php b/index.php
index 27772c9..1b1b037 100644
--- a/index.php
+++ b/index.php
@@ -17,7 +17,9 @@ $planets = [
 foreach ($planets as $planet) {
     try {
         echo $renderer->render($planet);
-    } catch (\Exception $e) {
+    } catch (\Exception\InvalidRadiusException $e) {
         echo 'Planet cannot be rendered';
+    } catch (\Exception\MissingHexException $e) {
+        echo 'Planet has no color';
     }
 }
