diff --git a/DeathStarII.php b/DeathStarII.php
index e69de29..9142b6b 100644
--- a/DeathStarII.php
+++ b/DeathStarII.php
@@ -0,0 +1,5 @@
+<?php
+
+class DeathStarII extends DeathStar
+{
+}
diff --git a/index.php b/index.php
index 0d1d771..8a3aee6 100644
--- a/index.php
+++ b/index.php
@@ -3,4 +3,4 @@
 require 'DeathStar.php';
 require 'DeathStarII.php';
 
-// set your $deathStar variable here
+$deathStar = new DeathStarII();
