diff --git a/PlanetInterface.php b/PlanetInterface.php
new file mode 100644
index 0000000..160b4d9
--- /dev/null
+++ b/PlanetInterface.php
@@ -0,0 +1,18 @@
+<?php
+
+interface PlanetInterface
+{
+    /**
+     * Return the radius if the planet, in thousands of kilometers.
+     *
+     * @return integer
+     */
+    public function getRadius();
+
+    /**
+     * Return the hex color (without the #) for the planet.
+     *
+     * @return string
+     */
+    public function getHexColor();
+}
diff --git a/index.php b/index.php
index 7ae3d33..4f0c671 100644
--- a/index.php
+++ b/index.php
@@ -1,6 +1,6 @@
 <?php
 
-require 'AbstractPlanet.php';
+require 'PlanetInterface.php';
 require 'SolidPlanet.php';
 require 'GasPlanet.php';
 require 'PlanetRenderer.php';
