diff --git a/src/Model/Planet/GasPlanet.php b/src/Model/Planet/GasPlanet.php
index cec9f65..cc82e1c 100644
--- a/src/Model/Planet/GasPlanet.php
+++ b/src/Model/Planet/GasPlanet.php
@@ -25,6 +25,11 @@ class GasPlanet implements PlanetInterface
         $this->mainElement = $mainElement;
     }
 
+    public function __destruct()
+    {
+        echo $this->name;
+    }
+
     public function __toString()
     {
         return $this->getName();
