diff --git a/index.php b/index.php
index 890d251..fd8a20a 100644
--- a/index.php
+++ b/index.php
@@ -4,16 +4,18 @@ spl_autoload_register(function ($class) {
     require __DIR__ . '/src/' . str_replace('\\', '/', $class) . '.php';
 });
 
-use Service\PlanetRenderer;
-use Model\Planet\GasPlanet;
 use Model\Planet\SolidPlanet;
-use Model\Planet\RandomlyColoredPlanet;
 
-$planet1 = new GasPlanet(GasPlanet::MATERIAL_AMMONIA, 49);
-$planet2 = new SolidPlanet(27, '353535');
-$planet3 = new RandomlyColoredPlanet('0969F9', 'F96909');
+$planet = new SolidPlanet(100, 'ff00ff', 12);
 
-$renderer = new PlanetRenderer();
-echo $renderer->render($planet1);
-echo $renderer->render($planet2);
-echo $renderer->render($planet3);
+// ...
+// just an example of how to use the DateTime object
+// $tomorrow = new DateTime();
+// $tomorrow->modify('+24 hours');
+
+?>
+
+<h3>
+    1 day from now:
+    <?php echo $tomorrow->format('Y-m-d H:i:s'); ?>
+</h3>
diff --git a/src/Model/Planet/SolidPlanet.php b/src/Model/Planet/SolidPlanet.php
index 161c20a..0d4a8a3 100644
--- a/src/Model/Planet/SolidPlanet.php
+++ b/src/Model/Planet/SolidPlanet.php
@@ -8,10 +8,13 @@ class SolidPlanet implements PlanetInterface
 
     private $hexColor;
 
-    public function __construct($radius, $hexColor)
+    private $dayLengthInHours;
+
+    public function __construct($radius, $hexColor, $dayLengthInHours = 24)
     {
         $this->radius = $radius;
         $this->hexColor = $hexColor;
+        $this->dayLengthInHours = $dayLengthInHours;
     }
 
     public function getRadius()
@@ -23,4 +26,9 @@ class SolidPlanet implements PlanetInterface
     {
         return $this->hexColor;
     }
+
+    public function getDayLengthInHours()
+    {
+        return $this->dayLengthInHours;
+    }
 }
