diff --git a/index.php b/index.php
index 34df129..6029a30 100644
--- a/index.php
+++ b/index.php
@@ -4,7 +4,9 @@ require __DIR__ . '/PlanetRenderer.php';
 require __DIR__ . '/PlanetInterface.php';
 require __DIR__ . '/RandomlyColoredPlanet.php';
 
-$planet = new \Model\Planet\RandomlyColoredPlanet('0969F9', 'F96909');
+use Model\Planet\RandomlyColoredPlanet;
+
+$planet = new RandomlyColoredPlanet('0969F9', 'F96909');
 
 $renderer = new PlanetRenderer();
 echo $renderer->render($planet);
