diff --git a/DeathStar.php b/DeathStar.php
index 98db915..2745432 100644
--- a/DeathStar.php
+++ b/DeathStar.php
@@ -2,5 +2,8 @@
 
 class DeathStar
 {
-    protected $planetarySuperLaserRange = 2000000;
+    public function getLaserRange()
+    {
+        return 2000000;
+    }
 }
diff --git a/DeathStarII.php b/DeathStarII.php
index 172ba06..9142b6b 100644
--- a/DeathStarII.php
+++ b/DeathStarII.php
@@ -2,11 +2,4 @@
 
 class DeathStarII extends DeathStar
 {
-    public function getSpecs()
-    {
-        return array(
-            'name' => 'DeathStarII',
-            'laser_range' => $this->planetarySuperLaserRange * 2,
-        );
-    }
 }
diff --git a/index.php b/index.php
index ed169b6..51edf31 100644
--- a/index.php
+++ b/index.php
@@ -7,12 +7,6 @@ $deathStar = new DeathStarII();
 
 ?>
 
-<h2>New DeathStar Specs</h2>
-<table>
-    <?php foreach ($deathStar->getSpecs() as $key => $val): ?>
-        <tr>
-            <th><?php echo $key; ?></th>
-            <td><?php echo $val; ?></td>
-        </tr>
-    <?php endforeach; ?>
-</table>
+<h3>
+    Laser Range <?php echo $deathStar->getLaserRange(); ?>
+</h3>
