diff --git a/index.php b/index.php
index a49c9d2..ca1c2bc 100644
--- a/index.php
+++ b/index.php
@@ -5,7 +5,7 @@ require __DIR__ . '/vendor/autoload.php';
 use Model\Planet\GasPlanet;
 use Service\PlanetRenderer;
 
-$planet = new GasPlanet(GasPlanet::MATERIAL_METHANE, 88);
+$planet = new GasPlanet('gold', 88);
 
 $renderer = new PlanetRenderer();
 echo $renderer->render($planet);
diff --git a/src/Model/Planet/GasPlanet.php b/src/Model/Planet/GasPlanet.php
index dce3a0e..bdc2c5b 100644
--- a/src/Model/Planet/GasPlanet.php
+++ b/src/Model/Planet/GasPlanet.php
@@ -16,6 +16,9 @@ class GasPlanet implements PlanetInterface
     public function __construct($mainElement, $diameter)
     {
         $this->diameter = $diameter;
+        if (!in_array($mainElement, $this->getAllElements())) {
+            throw new \Exception('This is not a valid element!');
+        }
         $this->mainElement = $mainElement;
     }
 
