diff --git a/index.php b/index.php
index 6893922..c35846d 100644
--- a/index.php
+++ b/index.php
@@ -19,7 +19,6 @@ $planets = [
     new GasPlanet('Uranus', GasPlanet::MATERIAL_METHANE, 51),
 ];
 $neptune = new GasPlanet('Neptune', GasPlanet::MATERIAL_HYDROGEN, 49);
-$neptune->name = 'Nep-Toon!'; // this is a silly name
 $planets[] = $neptune;
 
 $solarSystem = new SolarSystem($planets);
diff --git a/src/Model/Planet/GasPlanet.php b/src/Model/Planet/GasPlanet.php
index cc82e1c..660aa5d 100644
--- a/src/Model/Planet/GasPlanet.php
+++ b/src/Model/Planet/GasPlanet.php
@@ -4,13 +4,13 @@ namespace Model\Planet;
 
 class GasPlanet implements PlanetInterface
 {
+    use NameableItemTrait;
+
     const MATERIAL_AMMONIA = 'ammonia';
     const MATERIAL_HYDROGEN = 'hydrogen';
     const MATERIAL_HELIUM = 'helium';
     const MATERIAL_METHANE = 'methane';
 
-    private $name;
-
     private $diameter;
 
     private $mainElement;
@@ -35,13 +35,6 @@ class GasPlanet implements PlanetInterface
         return $this->getName();
     }
 
-    public function __set($name, $value)
-    {
-        if ('name' === $name) {
-            $this->name = $value;
-        }
-    }
-
     public static function getAllElements()
     {
         return [
@@ -52,11 +45,6 @@ class GasPlanet implements PlanetInterface
         ];
     }
 
-    public function getName()
-    {
-        return $this->name;
-    }
-
     public function getRadius()
     {
         return $this->diameter / 2;
diff --git a/src/Model/Planet/NameableItemTrait.php b/src/Model/Planet/NameableItemTrait.php
new file mode 100644
index 0000000..80e9d96
--- /dev/null
+++ b/src/Model/Planet/NameableItemTrait.php
@@ -0,0 +1,27 @@
+<?php
+
+namespace Model\Planet;
+
+trait NameableItemTrait
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @param string $name
+     */
+    public function setName($name)
+    {
+        $this->name = $name;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+}
diff --git a/src/Model/Planet/SolidPlanet.php b/src/Model/Planet/SolidPlanet.php
index e53196a..ca1f5c4 100644
--- a/src/Model/Planet/SolidPlanet.php
+++ b/src/Model/Planet/SolidPlanet.php
@@ -4,7 +4,7 @@ namespace Model\Planet;
 
 class SolidPlanet implements PlanetInterface
 {
-    private $name;
+    use NameableItemTrait;
 
     private $radius;
 
@@ -33,11 +33,6 @@ class SolidPlanet implements PlanetInterface
         return $tomorrow;
     }
 
-    public function getName()
-    {
-        return $this->name;
-    }
-
     public function getRadius()
     {
         return $this->radius;
