diff --git a/index.php b/index.php
index 6893922..c35846d 100644
--- a/index.php
+++ b/index.php
@@ -19,7 +19,6 @@ $planets = [
     new GasPlanet('Uranus', GasPlanet::MATERIAL_METHANE, 51),
 ];
 $neptune = new GasPlanet('Neptune', GasPlanet::MATERIAL_HYDROGEN, 49);
-$neptune->name = 'Nep-Toon!'; // this is a silly name
 $planets[] = $neptune;
 
 $solarSystem = new SolarSystem($planets);
diff --git a/src/Model/Planet/GasPlanet.php b/src/Model/Planet/GasPlanet.php
index cc82e1c..bcc919b 100644
--- a/src/Model/Planet/GasPlanet.php
+++ b/src/Model/Planet/GasPlanet.php
@@ -35,13 +35,6 @@ class GasPlanet implements PlanetInterface
         return $this->getName();
     }
 
-    public function __set($name, $value)
-    {
-        if ('name' === $name) {
-            $this->name = $value;
-        }
-    }
-
     public static function getAllElements()
     {
         return [
@@ -52,6 +45,11 @@ class GasPlanet implements PlanetInterface
         ];
     }
 
+    public function setName($name)
+    {
+        $this->name = $name;
+    }
+
     public function getName()
     {
         return $this->name;
diff --git a/src/Model/Planet/NameableItemTrait.php b/src/Model/Planet/NameableItemTrait.php
new file mode 100644
index 0000000..e69de29
diff --git a/src/Model/Planet/SolidPlanet.php b/src/Model/Planet/SolidPlanet.php
index e53196a..c13fbf3 100644
--- a/src/Model/Planet/SolidPlanet.php
+++ b/src/Model/Planet/SolidPlanet.php
@@ -33,6 +33,11 @@ class SolidPlanet implements PlanetInterface
         return $tomorrow;
     }
 
+    public function setName($name)
+    {
+        $this->name = $name;
+    }
+
     public function getName()
     {
         return $this->name;
